From 1fa0f088e9b016b7534837222770a859b929d56b Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Sat, 12 Dec 2020 16:25:24 -0800
Subject: [PATCH 754/767] octeontx2-sdp: Add pf_srn field in info exchange
 structure

commit eb54d2768783c5267e4f8f90871e124de00b7ebb from
git@git.assembla.com:cavium/WindRiver.linux.git

Add pf_srn field to notify the host about the starting ring number for
the PF. This is required in 98xx as it has two SDP blocks and each block
will have atleast 1 PF in the host. Also we start the ring number for
SDP1 from 128.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: If8be1af46d1bb01287528aba07270633307d15b6
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42127
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49038
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-sdp/sdp.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index b2650d1e371c..4319abfb2d7f 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -42,7 +42,8 @@ union ring {
 		u64 rpvf:8;
 		u64 rppf:8;
 		u64 numvf:8;
-		u64 rsvd:10;
+		u64 pf_srn:8;
+		u64 rsvd:2;
 		u64 raz:28;
 	} s;
 };
@@ -1176,6 +1177,11 @@ static int sdp_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	fw_rinfo.s.rppf = num_pf_rings;
 	fw_rinfo.s.rpvf = vf_rings[0];
 	fw_rinfo.s.numvf = max_vfs-1;
+	/*
+	 * For 98xx there are 2xSDPs so start the PF ring from 128 for SDP1
+	 * SDP0 has PCI revid 0 and SDP1 has PCI revid 1
+	 */
+	fw_rinfo.s.pf_srn = pdev->revision ? 128 : pf_srn;
 
 	dev_info(&pdev->dev, "Ring info 0x%llx\n", fw_rinfo.u);
 	writeq(fw_rinfo.u, sdp->sdp_base + SDPX_RINGX_IN_PKT_CNT(0));
-- 
2.31.1

